dali/base/dautils.cpp

```diff
@@ -3644,3 +3644,25 @@ void remapGroupsToDafilesrv(IPropertyTree *file, INamedGroupStore *resolver)
         }
     }
 }
+
+#ifdef NULL_DALIUSER_STACKTRACE
+static time_t lastNullUserLogEntry = (time_t)0;
+static CriticalSection nullUserLogCS;
+void logNullUser(IUserDescriptor * userDesc)
+{
+    StringBuffer userName;
+    if (userDesc)
+        userDesc->getUserName(userName);
+    if (nullptr == userDesc || userName.isEmpty())
+    {
+        CriticalBlock block(nullUserLogCS);
+        time_t timeNow = time(nullptr);
```

Comment:
discovery:  I was going to add a comment about using get_cycles_now() rather than time(), but decided to test it first.  Interestingly, modern implementations of time() are *very* efficient.  If any of these calls were in a time critical situation it would be more efficient to test the difference in time first.  I don't think that is the case, so simplest to leave as it is.

(see results in https://github.com/hpcc-systems/HPCC-Platform/pull/17327 for more details.)